tests/PHPStan/Rules/Functions/data/bug-6987.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types = 1);
+
+namespace Bug6987;
+
+class Foo {
+	public function doFoo()
+	{
+		$availableValues =  [
+			'ENABLED' => 123123123,
+			'DISABLED' => 555555,
+			'CANCELLED' => 11111,
+		];
+
+		$map = [];
+		foreach($availableValues as $key => $value){
+			$map[$this->transformKey(strtolower($key))] = $value;
```

Comment:
I was not able to reproduce [the error from the playground](https://phpstan.org/r/bf565b4a-39e9-4e59-8e74-7d83d5a9a4eb) with this test